src/pages/Auth/Auth.jsx

```diff
@@ -0,0 +1,73 @@
+import React, { useState } from "react";
+import { Link } from "react-router-dom";
+import { api } from "../../api/api";
+
+const Auth = () => {
+  const [userInfo, setUserInfo] = useState({
+    email: "",
+    password: "",
+  });
+  const { email, password } = userInfo;
+
+  const getUserInfo = (e) => {
+    const { name, value } = e.target;
+    setUserInfo({ ...userInfo, [name]: value });
+  };
+
+  const validation = email.includes("@") && password.length >= 8;
+
+  const signUp = async () => {
+    const response = await fetch(api.signUp, {
+      method: "POST",
+      headers: { "Content-type": "application/json;charset=utf-8" },
+      body: JSON.stringify(userInfo),
+    });
+    const createdUser = await response.json();
+    alert(createdUser.message);
+  };
+
+  return (
+    <div className="flex flex-col justify-center items-center h-screen max-w-md p-10 border-none">
+      <h1 className="mb-5 text-md">회원가입</h1>
+
+      <form className="flex flex-col">
+        <input
+          placeholder="ID"
+          type="email"
+          value={email}
+          name="email"
+          onChange={getUserInfo}
+          className={`p-3 px-4 border-2 w-[300px] ${
+            email.includes("@") && "border-emerald-200"
+          } rounded-md mb-5 focus:outline-none `}
```

Comment:
className에 너무 많은 관심이 쏠려있는 것 같습니다!!
이렇게 되면 rule에 대한 추가 요구사항이 들어왔을 때에 핸들링이 어려워 질 것 같아요.

rule에 대한 내용은 따로 util이나, 해당 컴포넌트 바깥에서 관리 하는건 어떨까요??

like this

1. 단순 분리
```ts
// 바깥쪽
const RULES = {
  id: (value) => value.includes('@'),
  password: (value) => value.length >= 8,
}
```

```suggestion
          className={`p-3 px-4 border-2 w-[300px] ${
            RULES.id(email) && "border-emerald-200"
          } rounded-md mb-5 focus:outline-none `}
```

2. Input Component 분리
그런데 작성하다보니 이 input은 사용자가 작성한 내용에 따라 보여지는 view가 달라집니다.
그래서 Core Input을 따로 만들고, LabeldInput 으로 Higher Ordering 하면 좋을 것 같아요!!
위에 대한 예시는 코드샌드박스 남겨놓습니다!!

https://codesandbox.io/s/input-jal-darugi-rwryx8?file=/src/App.js:0-34

Reply:
이렇게 하면 어떤 이점도 있냐면, 
input에 관련된 모든 RULE이 여기에 정리되어 있으니
1. 전체가 validate되었는지 한줄로 확인 할 수 있다.
```jsx
                                                   __👇함수__(함수호출)
Object.entries(userInfo).every(([key, value]) => RULES[key](value))
                                                       ¯¯¯   ¯¯¯¯¯
                              userInfo의 키값👆        👆Rules의 value로 넣을 값
```

2. 상황에 따라 filtering 할 수 있다 => RULE 확인을 안할 수 있다
```jsx
Object.entries(userInfo)
  .filter(필터로직)
  .every(확인로직)
```

Reply:
요거슨 auth 브랜치에서 해결해버렸습니다요

Reply:
전 갠적으로 react-hook-form에 yup 같은거 물려서 쓰는게 맞다고 봅니다 ㅋㅋㅋ

Reply:
react-hook-form, yup 한번 써보겠습니다..